test/e2e_node/gpus.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const acceleratorsFeatureGate = "Accelerators=true"
+
+// Serial because the test updates kubelet configuration.
+var _ = framework.KubeDescribe("GPU [Serial]", func() {
+	f := framework.NewDefaultFramework("gpu-test")
+	Context("attempt to use GPUs if available", func() {
+		It("setup the node and create pods to test gpus", func() {
+			By("ensuring that dynamic kubelet configuration is enabled")
+			enabled, err := isKubeletConfigEnabled(f)
+			framework.ExpectNoError(err)
+			if !enabled {
+				Skip("Dynamic Kubelet configuration is not enabled. Skipping test.")
+			}
+
+			By("enabling support for GPUs")
+			var oldCfg *componentconfig.KubeletConfiguration
```

Comment:
I probably didn't document it well enough then.  What is here works well for now.